corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/views/query.js

```diff
@@ -37,23 +61,61 @@ hqDefine("cloudcare/js/formplayer/menus/views/query", function () {
 
         ui: {
             submitButton: '#query-submit-button',
+            valueDropdown: 'select.query-field',
         },
 
         events: {
+            'change @ui.valueDropdown': 'changeDropdown',
             'click @ui.submitButton': 'submitAction',
         },
 
-        submitAction: function (e) {
-            e.preventDefault();
-            var payload = {};
-            var fields = $(".query-field");
-            var model = this.parentModel;
-            fields.each(function (index) {
+        getAnswers: function () {
+            var $fields = $(".query-field"),
+                answers = {},
+                model = this.parentModel;
+            $fields.each(function (index) {
                 if (this.value !== '') {
-                    payload[model[index].get('id')] = this.value;
+                    answers[model[index].get('id')] = this.value;
                 }
             });
-            FormplayerFrontend.trigger("menu:query", payload);
+            return answers;
+        },
+
+        changeDropdown: function (e) {
+            e.preventDefault();
+            var self = this;
+            var $fields = $(".query-field");
+
+            // If there aren't at least two dropdowns, there are no dependencies
+            if ($fields.filter("select").length < 2) {
+                return;
+            }
+
+            var Util = hqImport("cloudcare/js/formplayer/utils/util");
+            var urlObject = Util.currentUrlToObject();
+            urlObject.setQuery(this.getAnswers());
+            urlObject.setDoQuery(false);
+            var fetchingPrompts = FormplayerFrontend.getChannel().request("app:select:menus", urlObject);
+            $.when(fetchingPrompts).done(function (response) {
+                for (var i = 0; i < response.models.length; i++) {
+                    var choices = response.models[i].get('itemsetChoices');
+                    if (choices) {
+                        var $field = $($fields.get(i)),
+                            value = parseInt($field.val());
+                        $field.select2('close');    // force close dropdown, the set below can interfere with this when clearing selection
+                        self.collection.models[i].set({
```

Comment:
I also did not follow all of the var semantics, but there's a 1:1 mapping between `self.collection.models` and `response.models`?

Reply:
Yep, the models are the search fields, and this request is checking formplayer to see if updating one dropdown's value caused any other dropdown's options to update. The response always includes all fields.